third_party/nvfuser/csrc/python_frontend/fusion_record.h

```diff
@@ -1207,12 +1237,59 @@ struct OutputRecord : RecordFunctor {
     } else {
       // With C++17, this statement should be "if constexpr"
       if (std::is_same<OutputType, TensorView>::value) {
-        fd.addOutput(output->template as<TensorView>());
+        auto tv_output = output->template as<TensorView>();
+
+        if (!stride_order_.empty()) {
+          std::vector<int64_t> reverse_perm(stride_order_.size());
+          size_t duplicate_check = 0;
+          for (const auto i : c10::irange(stride_order_.size())) {
+            TORCH_CHECK(
+                stride_order_[i] >= 0 &&
+                    stride_order_[i] < (int64_t)reverse_perm.size(),
+                "stride_order elements need to be within [0, stride_order.size())!");
+            reverse_perm[stride_order_[i]] = i;
+            duplicate_check |= 1 << stride_order_[i];
+          }
+          TORCH_CHECK(
+              duplicate_check == (1 << reverse_perm.size()) - 1,
+              "duplicated elements in stride_order detected!");
+          tv_output = permute(tv_output, reverse_perm);
```

Comment:
I could be wrong but I think this call is first time the length of provided perm is checked to equal ndim of other argument. It might be nice to check that right up front.

Reply:
good point. I had it in `python_bindings.cpp` earlier, and removed that as duplication.

But I think it makes sense to throw an error up front. I'll add one down there.